ts/main.ts

```diff
@@ -255,12 +307,34 @@ $header.addEventListener('click', (event: Event) => {
 
 $cardList.addEventListener('click', (event: Event) => {
   const $eventTarget = event.target as HTMLElement;
-  if ($eventTarget.closest('.card-list > .card')) {
+  if (
+    $eventTarget.closest('.card-list > .card') &&
+    $eventTarget.tagName !== 'I'
+  ) {
     const $card = $eventTarget.closest('.card') as HTMLElement;
     if ($card.dataset.baseId) {
       const cardBaseId = $card.dataset.baseId;
-      viewSwap('exercise-details');
       populateExerciseDetails(parseInt(cardBaseId));
+      viewSwap('exercise-details');
+    }
+  }
+});
+
+document.addEventListener('click', (event: Event) => {
+  const $eventTarget = event.target as HTMLElement;
+  if ($eventTarget.tagName !== 'I') return;
+  if ($eventTarget.closest('.card-list > .card')) {
```

Comment:
You should try to specify where events need to happen, using delegation to determine what the highest level container specifically is needed to check. The problem with having it on your `document` is that this function runs anytime a user clicked anywhere on your document which you don't really want to do